pkg/scheduler/metrics/source/metrics_client_prometheus.go

```diff
@@ -37,17 +39,33 @@ const (
 
 type PrometheusMetricsClient struct {
 	address string
+	conf    map[string]string
 }
 
-func NewPrometheusMetricsClient(address string) (*PrometheusMetricsClient, error) {
-	return &PrometheusMetricsClient{address: address}, nil
+func NewPrometheusMetricsClient(address string, conf map[string]string) (*PrometheusMetricsClient, error) {
+	return &PrometheusMetricsClient{address: address, conf: conf}, nil
 }
 
 func (p *PrometheusMetricsClient) NodeMetricsAvg(ctx context.Context, nodeName string, period string) (*NodeMetrics, error) {
 	klog.V(4).Infof("Get node metrics from Prometheus: %s", p.address)
-	client, err := api.NewClient(api.Config{
-		Address: p.address,
-	})
+	var client api.Client
+	var err error
+	if strings.Contains(p.address, "https") {
```

Comment:
Here, as before, it is recommended to separate configuration parsing, structure preparation and final client creation.
1. Avoid duplication of code
2. It is easier to add new configurations later

Reply:
@wangyang0616 fixed